DOC: update release guide (#13421)
matplotlib/matplotlib: doc/devel/release_guide.rst

.. _release_chkdocs:

> Check and updating Docs

Reviewer: Verb tense mismatch.

-----------------------

Before tagging, update the "what's new" and API changes listings.

 1. copy the current content to a file in :file:`doc/users/prev_whats_new`

>  2. merging all of the files in :file:`doc/users/next_whats_new/` into

Reviewer: Verb tense mismatch.
```suggestion
 2. merge all of the files in :file:`doc/users/next_whats_new/` into
```

Review and commit changes.  Some issue/PR titles may not be valid rst (the most common issue is
``*`` which is interpreted as unclosed markup).

.. note

   Make sure you authenticate against the github API (either via
   keyring or via putting an oauth token in :file:`~/.ghoauth`).  If you
   do not you will get blocked by github for going over the API rate
   limits.

.. _release_chkdocs:

Check and updating Docs
-----------------------

Before tagging, update the "what's new" and API changes listings.

 1. copy the current content to a file in :file:`doc/users/prev_whats_new`
 2. merging all of the files in :file:`doc/users/next_whats_new/` into

>     :file:`doc/users/whats_new.rst` and deleting the individiual files

Reviewer: ```suggestion
    :file:`doc/users/whats_new.rst` and delete the individual files
```

 3. comment out the next whats new glob at the top

Similarly for the API changes

 1. copy the current api changes to a file is :file:`doc/api/prev_api_changes`
 2. merge all of the files in :file:`doc/api/next_api_changes/` into
    :file:`doc//whats_new.rst`
 3. comment out the next API changes at the top.

In both cases step 3 will have to be un-done right after the release.

Finally, make sure that the docs build cleanly ::

>   make -Cdoc O-n=$(nproc) html latexpdf

Reviewer: That doesn't look right; it's definitely `O=`.

Author: The new version runs, the old version gives me weird errors...

```
jupiter@00:01  ➤  make -Cdoc O=n-$(nproc) html latexpdf
make: Entering directory '/home/tcaswell/source/p/matplotlib/doc'
usage: __main__.py [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
__main__.py: error: unrecognized arguments: n-8
make: *** [Makefile:20: html] Error 2
make: Leaving directory '/home/tcaswell/source/p/matplotlib/doc'
```

vs

```
(py37) ✘  ~/source/p/matplotlib [doc_backports ✭|⚑ 8]
jupiter@00:01  ➤  make -Cdoc O-n=$(nproc) html latexpdf
make: Entering directory '/home/tcaswell/source/p/matplotlib/doc'
Running Sphinx v1.8.4
loading pickled environment... done
[autosummary] generating autosummary 
....
```

Reviewer: The "old" version has a typo and should be `O=-n$(nproc)`; the new version passes a different variable and does nothing.

Author: Sorry, miss-type

```
(py37) ✘  ~/source/p/matplotlib [doc_release_notes ✭|⚑ 8]
jupiter@00:15  ➤  make -Cdoc O=-n$(nproc) html latexpdf
make: Entering directory '/home/tcaswell/source/p/matplotlib/doc'
usage: __main__.py [OPTIONS] SOURCEDIR OUTPUTDIR [FILENAMES...]
__main__.py: error: argument -n: ignored explicit argument '8'
make: *** [Makefile:30: html] Error 2
make: Leaving directory '/home/tcaswell/source/p/matplotlib/doc'
```

Author: Because it should be `-j` not `-n` https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-j

Author: Which of course breaks things now that it is actually working....

```
Warning, treated as error:
the sphinx_copybutton extension does not declare if it is safe for parallel reading, assuming it isn't - please ask the extension author to check and make it explicit
```

Author: Which is fixed on their master branch....

Reviewer: Ah, okay, must have been a copy-paste bug from the `make.py` switch.

Author: copybutton did a new release for us (thank @choldgraf !)

We distribute mac, windows, and many linux wheels as well as a source
tarball via pypi.  Before uploading anything, contact the various
builders.  Mac and manylinux wheels are built on travis .  You need to
edit the :file:`.travis.yml` file and push to the correct branch of
`the build project
<https://github.com/MacPython/matplotlib-wheels>`__.  For new minor

> version create a new branch, for bug-fixes continue to use the current

Reviewer: ```suggestion
versions create a new branch, for bug-fixes continue to use the current
```

release branch.

The auto-tick bot should open a pull request into the `conda-forge
feedstock <https://github.com/conda-forge/matplotlib-feedstock>`__.
Review and merge (if you have the power to).

> .. warning

Reviewer 2: ```suggestion
.. warning::
```

- matplotlib-announce@python.org

For final releases announcements should also be sent to the
numpy/scipy/scikit-image mailing lists.

In addition, announcements should be made on social networks (twitter
via the ``@matplotlib`` account, any other via personal accounts).

> `NumFOCUS <http://www.numfocus.org/>`__ should be contacted for

Reviewer: https://numfocus.org/ now.
